minifi-commons/minifi-commons-schema/src/main/java/org/apache/nifi/minifi/commons/schema/v2/CorePropertiesSchemaV2.java

```diff
@@ -0,0 +1,95 @@
+
+/**
+ *
+ */
+public class CorePropertiesSchemaV2 extends BaseSchema implements WritableSchema {
+
+    public static final String FLOW_CONTROLLER_SHUTDOWN_PERIOD_KEY = "flow controller graceful shutdown period";
+    public static final String FLOW_SERVICE_WRITE_DELAY_INTERVAL_KEY = "flow service write delay interval";
+    public static final String ADMINISTRATIVE_YIELD_DURATION_KEY = "administrative yield duration";
+    public static final String BORED_YIELD_DURATION_KEY = "bored yield duration";
+
+    public static final String DEFAULT_FLOW_CONTROLLER_GRACEFUL_SHUTDOWN_PERIOD = "10 sec";
+    public static final String DEFAULT_FLOW_SERVICE_WRITE_DELAY_INTERVAL = "500 ms";
+    public static final String DEFAULT_ADMINISTRATIVE_YIELD_DURATION = "30 sec";
+    public static final String DEFAULT_BORED_YIELD_DURATION = "10 millis";
+    public static final int DEFAULT_MAX_CONCURRENT_THREADS = 1;
+
+    private String flowControllerGracefulShutdownPeriod = DEFAULT_FLOW_CONTROLLER_GRACEFUL_SHUTDOWN_PERIOD;
+    private String flowServiceWriteDelayInterval = DEFAULT_FLOW_SERVICE_WRITE_DELAY_INTERVAL;
+    private String administrativeYieldDuration = DEFAULT_ADMINISTRATIVE_YIELD_DURATION;
+    private String boredYieldDuration = DEFAULT_BORED_YIELD_DURATION;
+    private Number maxConcurrentThreads = DEFAULT_MAX_CONCURRENT_THREADS;
+
+    public CorePropertiesSchemaV2() {
+    }
+
+    public CorePropertiesSchemaV2(Map map) {
+        flowControllerGracefulShutdownPeriod = getOptionalKeyAsType(map, FLOW_CONTROLLER_SHUTDOWN_PERIOD_KEY, String.class,
+                CORE_PROPS_KEY, DEFAULT_FLOW_CONTROLLER_GRACEFUL_SHUTDOWN_PERIOD);
+        flowServiceWriteDelayInterval = getOptionalKeyAsType(map, FLOW_SERVICE_WRITE_DELAY_INTERVAL_KEY, String.class,
+                CORE_PROPS_KEY, DEFAULT_FLOW_SERVICE_WRITE_DELAY_INTERVAL);
+        administrativeYieldDuration = getOptionalKeyAsType(map, ADMINISTRATIVE_YIELD_DURATION_KEY, String.class,
+                CORE_PROPS_KEY, DEFAULT_ADMINISTRATIVE_YIELD_DURATION);
+        boredYieldDuration = getOptionalKeyAsType(map, BORED_YIELD_DURATION_KEY, String.class, CORE_PROPS_KEY, DEFAULT_BORED_YIELD_DURATION);
+        maxConcurrentThreads = getOptionalKeyAsType(map, MAX_CONCURRENT_THREADS_KEY, Number.class,
+                CORE_PROPS_KEY, DEFAULT_MAX_CONCURRENT_THREADS);
+    }
+
+    @Override
+    public Map<String, Object> toMap() {
+        Map<String, Object> result = mapSupplier.get();
+        result.put(FLOW_CONTROLLER_SHUTDOWN_PERIOD_KEY, flowControllerGracefulShutdownPeriod);
+        result.put(FLOW_SERVICE_WRITE_DELAY_INTERVAL_KEY, flowServiceWriteDelayInterval);
+        result.put(ADMINISTRATIVE_YIELD_DURATION_KEY, administrativeYieldDuration);
+        result.put(BORED_YIELD_DURATION_KEY, boredYieldDuration);
+        result.put(MAX_CONCURRENT_THREADS_KEY, maxConcurrentThreads);
+        return result;
+    }
+
+    public String getFlowControllerGracefulShutdownPeriod() {
```

Comment:
The getters should be able to be safely deleted as this V2 class will just be used as an intermediary during conversion